### DISCUSSION TITLE
Amazing Obsidian Plugin - one of the best! ❤

### DISCUSSION THREAD

**Discussion by ManasMadrecha** [aptend/typing-transformer-obsidian#58] [Show and tell]
## Super Easy to use 

While the documentation on the README page seems a bit daunting, this plugin is actually really very simple to use. 

## Obsidian writing now similar to MS Word

- Using a simple MS Word macro (found on [YouTube](https://www.youtube.com/watch?v=bG-HCHzAAUQ)), I got the list of AutoCorrect word options in my MS Word. 
- Copied them into Excel
- Using Excel's simple Concatenate, I converted those 1000 words into the requisite format and pasted this long list into the Plugin settings.
- And that's it!
- Now, I can use the power of MS Word in Obsidian. 

## Highly customizable

I even added some customized words based on the chat language, e.g., by simply typing `whr r u`, I get `where are you`. (did this by adding these lines in the settings 
```
' u |' -> ' you |'
' r |' -> ' are |'
```

The best thing about this Plugin is that you are in total control. E.g., by simply typing `u`, if `you` came, then it would not be good. But, if you are adding a space, you can be absolutely sure that you want to treat `u` like a whole word, and so convert it. Also, converting every small `i` to `I` is not desirable, because it will convert even `this` to `thIs`. So, you can add a rule that
```
' i |' -> ' I |'
```

Also, by simply placing `|` anywhere we want, we can get the cursor anywhere we want. This is so easy to understand even for non-coders. 
```
'==|' -> '==|==' #highlight text
```

Here the mark-up is added, and the cursor is placed in between the mark-up for to you continue writing seamlessly, without needing to go manually back to the middle.

## Improved typing speed

How Google keyboard on Android mobile handles our errors and makes our typing speed so so fast just by swiping here and there, similarly you can type absolutely any typo on now Desktop now and be rest assured that this plugin will take care of all the errors. Just invest some time in knowing your common typo, and add them to the Plugin.  

This plugin have added another amazing cherry on the cake of Obsidian. 
Stay Blessed ❤

**Comment by aptend** [comment]
Whoa, I am so glad the Plugin can help you! I didn't even think that the Plugin can be used in a way like AutoCorrect, that's cool.

And I feel your post is more clear and readable, maybe there should be fewer tech details and more user cases in the README…… If you are interested, please feel free to create a PR for the README ☺️ .

Another little question, did you notice an obvious CPU usage increase working with those 1000+ rules? If so I probably need to do some optimization. The discussion tab is enabled now, maybe you could share your auto-correct settings there, then I can have a test on my machine.

Thanks for your feedback, appreciate it~

**Comment by ManasMadrecha** [comment]
I didn't notice the CPU usage, but yes, adding thousands of words made the Obsidian typing a bit janky, so yesterday, I trimmed them to only my own custom Autocorrect list (50--100), instead of having the huge default MS Word list. This has made the typing more predictable and still super fast.

So, yes, **as of now, adding 1000+ list is not advisable** unless optimization is done in the plugin. You are most welcome.

**Comment by aptend** [comment]
Hi @ManasMadrecha, I have optimized the lookup process for a rule in version 0.2.1, theoretically, the experience should be significantly improved.  So you can give it a try, let me know if any problems
